task/06-conditions-n-loops-tasks.js

```diff
@@ -353,7 +364,30 @@ function isBracketsBalanced(str) {
  *
  */
 function timespanToHumanString(startDate, endDate) {
-    throw new Error('Not implemented');
+    let time = endDate.getTime() - startDate.getTime(),
+        int = new Date(time),
+        seconds = int.getUTCSeconds() + int.getUTCMilliseconds()/1000,
+        minutes = int.getUTCMinutes() + seconds/60,
+        hours = int.getUTCHours() + minutes/60,
+        days = (int.getUTCFullYear() - 1970)*365 + int.getUTCMonth()*30.42 + (int.getUTCDate() - 1) + hours/24,
+        str;
+
+    function round(value) {
+        let x = 1.0 / -1; return Math.round(value * x) / x;
+    }
+
+    (days    >= 546 )  ? str = round(days/365) + " years ago" :
+    (days    > 345)    ? str = "a year ago" : 
+    (days    >= 45)    ? str = round(days/30) + " months ago":
+    (days    > 25)     ? str = "a month ago" :
+    (days    > 1.5 )   ? str = round(days) + " days ago" :
+    (days*24 > 22 )    ? str = "a day ago" :
+    (hours   > 1.5 )   ? str = round(hours) + " hours ago" :
+    (hours   > 0.75 )  ? str = "an hour ago" :
+    (minutes > 1.5)    ? str = round(minutes) + " minutes ago" :
+    (minutes > 0.75 )  ? str = "a minute ago" :
+    (seconds > 0 )     ? str = "a few seconds ago" : 0;
```

Comment:
avoid this code style, if-else is more readable than multiple conditional operators ? :